### DISCUSSION TITLE
Cache Memory Usage

### DISCUSSION THREAD

**Discussion by amcheng** [nielsbasjes/yauaa#776] [Q&A] [answered]
Hello Anyone,

About how much memory does the included Caffeine cache use per cached user agent?

I did some rough estimations using a heap dump and I calculated about 1.7 kB per user agent. Does that sound right?

Thanks,
Alex

**Comment by nielsbasjes** [comment, marked as answer]
I haven't tested this and this sounds about right.

The cache holds a complete parsed UserAgent which is all keys and values of both the input and output + overhead for the classes and indexing.

I took two of the testcases from the yaml to do a quick estimate just based on the textual content that must be stored:
- Basic simple Chrome on Android: ~ 800 bytes
- Full example with all client hints: ~1800 bytes. 

Add overhead of the Java classes to that and I expect the caching to need something in the 1-2KiB per record range.
So your 1.7KiB sounds right to me.

**Reply by amcheng** [reply]
Thank you! We appreciate your answer - we are an industry user of your library.
